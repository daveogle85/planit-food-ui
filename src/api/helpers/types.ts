export enum RestVerb {
  GET = 'GET',
  POST = 'POST',
  PUT = 'PUT',
  DELETE = 'DELETE',
  OPTIONS = 'OPTIONS',
}

export type Options = {
  method: RestVerb;
  redirect: 'follow' | 'error' | 'manual' | undefined;
  headers: Headers;
  body?: string;
};
