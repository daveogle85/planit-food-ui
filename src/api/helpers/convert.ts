import { v4 as uuid } from 'uuid';

import { ApiDish, Dish } from '../types/DishTypes';
import { ApiMeal, Meal } from '../types/MealTypes';
import { List, ApiList } from '../types/ListTypes';
import { ApiDay, Day } from '../types/DayTypes';

export const convertFromDishApi = (dish: ApiDish): Dish => ({
  ...dish,
  localId: uuid(),
});

export const convertFromMealApi = (meal: ApiMeal): Meal => {
  const dishes = meal.dishes?.map(d => {
    const dish = convertFromDishApi(d);
    return dish;
  });
  return { ...meal, localId: uuid(), dishes };
};

export const convertFromDayApi = (day: ApiDay): Day => {
  const meal = convertFromMealApi(day.meal);
  return { ...day, meal };
};

export const convertToApiDish = (dish: Dish): ApiDish => {
  const { localId, ...apiDish } = dish;
  return apiDish;
};

export const convertToLightApiDish = (dish: Dish): ApiDish => {
  const { id, name, notes, dishType } = dish;
  return {
    id,
    name,
    notes,
    dishType,
  };
};

export const convertToApiMeal = (meal: Meal): ApiMeal => {
  const { localId, ...apiMeal } = meal;
  return {
    ...apiMeal,
    dishes: (meal.dishes ? meal.dishes.map(convertToApiDish) : []) as ApiDish[],
  };
};

export const convertToLightApiMeal = (meal: Meal): ApiMeal => {
  const { id, name, dishes, notes } = meal;
  const updatedDishes = dishes?.map(convertToLightApiDish);
  return {
    id,
    name,
    dishes: updatedDishes,
    notes,
  };
};

export const convertToListApiRequest = (list: List): ApiList => {
  const updatedMeals = list.meals.map(convertToLightApiMeal);
  return {
    ...list,
    meals: updatedMeals,
  };
};

export const convertToLightApiDay = (day: Day): ApiDay => {
  const meal = convertToLightApiMeal(day.meal);
  return { ...day, date: day.date!, meal };
};

export const convertToApiDay = (day: Day): ApiDay => {
  const meal = convertToApiMeal(day.meal);
  return { ...day, date: day.date!, meal };
};
