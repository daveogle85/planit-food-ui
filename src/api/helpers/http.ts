async function fetchWithTimeout(
  url: string,
  options = {},
  timeout = 5000
): Promise<Response> {
  const controller = new AbortController();
  const { signal } = controller;
  return new Promise((resolve, reject) => {
    const timer = setTimeout(() => {
      reject(new Error(`Timeout for request to ${url}`));
      controller.abort();
    }, timeout);
    fetch(url, { signal, ...options })
      .finally(() => clearTimeout(timer))
      .then(resolve, reject);
  });
}

export async function get<T>(path: string, accessToken: string): Promise<T> {
  try {
    const response = await fetchWithTimeout(
      `${process.env.REACT_APP_SERVER_BASE_URL}${path}`,
      {
        method: 'GET',
        headers: new Headers({
          'Content-Type': 'application/json',
          authorization: `Bearer ${accessToken}`,
        }),
      }
    );
    return response.json();
  } catch (ex) {
    console.error(ex.message);
    throw ex;
  }
}

// async function post<T>(url: string, body: T) {
//   try {
//     const { getIdTokenClaims } = useAuth0();
//     const accessToken = await getIdTokenClaims();
//     const response = await fetch(
//       `${process.env.REACT_APP_SERVER_BASE_URL}/blog/post`,
//       {
//         method: 'post',
//         headers: new Headers({
//           'Content-Type': 'application/json',
//           Accept: 'application/json',
//           authorization: `Bearer ${accessToken.__raw}`,
//         }),
//         body: JSON.stringify(body),
//       }
//     );
//     return response.ok;
//   } catch (ex) {
//     return false;
//   }
// }
