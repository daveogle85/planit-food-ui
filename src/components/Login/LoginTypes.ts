import { History } from 'history';

export type LoginProps = {
  history: History;
};
