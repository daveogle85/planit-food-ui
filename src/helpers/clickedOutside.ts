import React, { useEffect } from 'react';

/**
 * Hook that handles clicks outside of the passed ref
 */
function useOutsideAlerter(
  ref: React.MutableRefObject<null>,
  onClickOutside: () => void
) {
  useEffect(() => {
    function handleClickOutside(event: Event) {
      if (ref?.current && !(ref.current as any).contains(event.target)) {
        onClickOutside();
      }
    }

    // Bind the event listener
    document.addEventListener('mousedown', handleClickOutside);
    return () => {
      // Unbind the event listener on clean up
      document.removeEventListener('mousedown', handleClickOutside);
    };
  }, [ref, onClickOutside]);
}

export default useOutsideAlerter;
