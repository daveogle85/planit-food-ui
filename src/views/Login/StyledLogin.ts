import { centerFlex, fullScreen, pointer } from '../../styles/common';
import { loginButtonDim, loginModalDim } from '../../styles/heights';
import { bounce } from '../../styles/keyframes';
import styled from '../../styles/theme';

export const styledLogin = (Login: React.ComponentType) => styled(Login)`
  background-color: ${props => props.theme.colours.background.logoBlue};
  ${centerFlex}
  ${fullScreen}
  .login-window {
    background-color: ${props => props.theme.colours.background.backgroundGrey};
    border-radius: ${props => props.theme.border.radius.medium};
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: ${props => props.theme.spacing.medium};
    flex-direction: column;
    width: ${loginModalDim.width};
    height: ${loginModalDim.height};

    .logo-img {
      max-width: 200px;
    }

    .login-action {
      ${centerFlex};
      flex-direction: column;
    }

    .text {
      color: ${props => props.theme.colours.background.backgroundWhite};
      margin-bottom: ${props => props.theme.spacing.small};
    }
  }
`;

export const styledLoginButton = (LoginButton: React.ComponentType) => styled(
  LoginButton
)`
  *,
  *:before,
  *:after {
    box-sizing: border-box;
    outline: none;
  }

  ${pointer};
  position: relative;
  ${centerFlex}
  width: ${loginButtonDim.width};
  height: ${loginButtonDim.height};
  background-color: ${props => props.theme.colours.background.logoBlue};
  border-radius: ${props => props.theme.border.radius.medium};
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
  transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);
  overflow: hidden;
  &:before,
  &:after {
    position: absolute;
    top: 0;
    display: flex;
    align-items: center;
    width: 50%;
    height: 100%;
    transition: 0.25s linear;
    z-index: ${props => props.theme.zIndex.one};
  }
  &:before {
    content: '';
    left: 0;
    justify-content: flex-end;
    background-color: ${props => props.theme.colours.background.logoBlue};
  }
  &:after {
    content: '';
    right: 0;
    justify-content: flex-start;
    background-color: ${props => props.theme.colours.background.logoBlue};
  }
  &:hover {
    background-color: ${props => props.theme.colours.white};
    box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22);
    span {
      opacity: 0;
      z-index: -3;
    }
    &:before {
      opacity: 0.5;
      transform: translateY(-100%);
    }
    &:after {
      opacity: 0.5;
      transform: translateY(100%);
    }
  }
  span {
    position: absolute;
    top: 0;
    left: 0;
    ${centerFlex}
    width: 100%;
    height: 100%;
    color: ${props => props.theme.colours.background.whiteSmoke};
    font-size: ${props => props.theme.font.size.small};
    font-weight: ${props => props.theme.font.weight.bold};
    opacity: 1;
    transition: opacity 0.25s;
    z-index: ${props => props.theme.zIndex.two};
  }
  .social-link {
    cursor: pointer;
    position: relative;
    ${centerFlex}
    width: 100%;
    height: 100%;
    color: ${props => props.theme.colours.background.whiteSmoke};
    text-decoration: none;
    transition: 0.25s;
    svg {
      width: 20%;
      height: 100%;
      transform: scale(1);
    }
    &:hover {
      background-color: rgba(
        ${props => props.theme.colours.background.whiteSmoke},
        0.1
      );
      animation: ${bounce} 0.4s linear;
    }
  }
`;
