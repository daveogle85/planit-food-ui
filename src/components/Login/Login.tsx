import React from 'react';
import { useDispatch } from 'react-redux';
import { withRouter } from 'react-router-dom';

import { useAuth0 } from '../../contexts/auth0-context';
import { setAuth } from '../../ducks/auth/AuthReducer';
import Google from '../../images/google';
import { EmotionProps } from '../../styles/types';
import { LoginProps } from './LoginTypes';
import { styledLogin, styledLoginButton } from './StyledLogin';

export const Login: React.FC<LoginProps> = (props: LoginProps) => {
  const dispatch = useDispatch();
  const { loginWithRedirect, loading, isAuthenticated } = useAuth0();

  if (isAuthenticated) {
    props.history.push('/');
  }

  const handleClick = () => {
    loginWithRedirect({});
    dispatch(setAuth(true));
  };

  const LoginButtonRaw: React.FC<EmotionProps> = props => (
    <div className={props.className}>
      <span>Login With Google</span>
      <div className="social-link" onClick={handleClick}>
        <Google />
      </div>
    </div>
  );

  const LoginButton = styledLoginButton(LoginButtonRaw);

  if (loading) {
    return <div>Loading...</div>;
  }

  return (
    <div className={props.className}>
      <div className="login-window">
        <div className="text">Please login to continue</div>
        <LoginButton />
      </div>
    </div>
  );
};

export default styledLogin(withRouter(Login));
