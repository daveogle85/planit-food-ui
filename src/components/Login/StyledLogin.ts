import { fullScreen } from '../../styles/common';
import styled from '../../styles/theme';
import Login from './Login';

export default styled(Login)`
  background-color: ${props => props.theme.colours.background.backgroundGrey};
  display: flex;
  align-items: center;
  justify-content: center;
  ${fullScreen}
  .login-window {
    background-color: ${props =>
      props.theme.colours.background.backgroundWhite};
    border-radius: ${props => props.theme.border.radius.medium};
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 300px;
    height: 500px;

    .text {
      margin-bottom: ${props => props.theme.spacing.large};
    }
  }
`;
