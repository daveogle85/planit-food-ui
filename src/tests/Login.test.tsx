import serializer from 'jest-emotion';

import { Login } from '../views/Login/Login';
import { renderWithTheme } from './helpers';
import React from 'react';
import { styledLogin } from '../views/Login/StyledLogin';
jest.mock('../contexts/auth0-context', () => ({
  useAuth0: jest.fn().mockReturnValue({
    loginWithRedirect: jest.fn(),
    loading: false,
    isAuthenticated: false,
  }),
}));

expect.addSnapshotSerializer(serializer);
test('Login renders correctly', () => {
  const TestLogin = styledLogin(Login as any);
  const button = renderWithTheme(<TestLogin />).toJSON();
  expect(button).toMatchSnapshot();
});
