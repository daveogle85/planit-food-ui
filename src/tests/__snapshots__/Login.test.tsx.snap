// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
@keyframes animation-0 {
  40% {
    -webkit-transform: scale(1.4);
    -ms-transform: scale(1.4);
    transform: scale(1.4);
  }

  60% {
    -webkit-transform: scale(0.8);
    -ms-transform: scale(0.8);
    transform: scale(0.8);
  }

  80% {
    -webkit-transform: scale(1.2);
    -ms-transform: scale(1.2);
    transform: scale(1.2);
  }

  100% {
    -webkit-transform: scale(1.0);
    -ms-transform: scale(1.0);
    transform: scale(1.0);
  }
}

.emotion-1 {
  background-color: #2b2d2f;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
}

.emotion-1 .login-window {
  background-color: #fffff7;
  border-radius: 0.3em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 300px;
  height: 500px;
}

.emotion-1 .login-window .text {
  margin-bottom: 30px;
}

.emotion-0 {
  cursor: pointer;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 130px;
  height: 30px;
  background-color: #4db6ac;
  border-radius: 7px;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);
  -webkit-transition: all 0.3s cubic-bezier(0.25,0.8,0.25,1);
  transition: all 0.3s cubic-bezier(0.25,0.8,0.25,1);
  overflow: hidden;
}

.emotion-0 *,
.emotion-0 *:before,
.emotion-0 *:after {
  box-sizing: border-box;
  outline: none;
}

.emotion-0:before,
.emotion-0:after {
  position: absolute;
  top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 50%;
  height: 100%;
  -webkit-transition: 0.25s linear;
  transition: 0.25s linear;
  z-index: 1;
}

.emotion-0:before {
  content: '';
  left: 0;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  background-color: #4db6ac;
}

.emotion-0:after {
  content: '';
  right: 0;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: #4db6ac;
}

.emotion-0:hover {
  background-color: #46627f;
  box-shadow: 0 14px 28px rgba(0,0,0,0.25),0 10px 10px rgba(0,0,0,0.22);
}

.emotion-0:hover span {
  opacity: 0;
  z-index: -3;
}

.emotion-0:hover:before {
  opacity: 0.5;
  -webkit-transform: translateY(-100%);
  -ms-transform: translateY(-100%);
  transform: translateY(-100%);
}

.emotion-0:hover:after {
  opacity: 0.5;
  -webkit-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
}

.emotion-0 span {
  position: absolute;
  top: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  color: #f5f5f5;
  font-size: 12px;
  font-weight: 700;
  opacity: 1;
  -webkit-transition: opacity 0.25s;
  transition: opacity 0.25s;
  z-index: 2;
}

.emotion-0 .social-link {
  cursor: pointer;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  color: #f5f5f5;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: 0.25s;
  transition: 0.25s;
}

.emotion-0 .social-link svg {
  width: 20%;
  height: 100%;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.emotion-0 .social-link:hover {
  background-color: rgba( #f5f5f5, 0.1 );
  -webkit-animation: animation-0 0.4s linear;
  animation: animation-0 0.4s linear;
}

<div
  className="emotion-1"
>
  <div
    className="login-window"
  >
    <div
      className="text"
    >
      Please login to continue
    </div>
    <div
      className="emotion-0"
    >
      <span>
        Login With Google
      </span>
      <div
        className="social-link"
        onClick={[Function]}
      >
        <svg
          height="1792"
          viewBox="0 0 1792 1792"
          width="1792"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M896 786h725q12 67 12 128 0 217-91 387.5T1282.5 1568 896 1664q-157 0-299-60.5T352 1440t-163.5-245T128 896t60.5-299T352 352t245-163.5T896 128q300 0 515 201l-209 201q-123-119-306-119-129 0-238.5 65T484 652.5 420 896t64 243.5T657.5 1316t238.5 65q87 0 160-24t120-60 82-82 51.5-87 22.5-78H896V786z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Login renders correctly 1`] = `
@keyframes animation-0 {
  40% {
    -webkit-transform: scale(1.4);
    -ms-transform: scale(1.4);
    transform: scale(1.4);
  }

  60% {
    -webkit-transform: scale(0.8);
    -ms-transform: scale(0.8);
    transform: scale(0.8);
  }

  80% {
    -webkit-transform: scale(1.2);
    -ms-transform: scale(1.2);
    transform: scale(1.2);
  }

  100% {
    -webkit-transform: scale(1.0);
    -ms-transform: scale(1.0);
    transform: scale(1.0);
  }
}

.emotion-1 {
  background-color: #2b2d2f;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
}

.emotion-1 .login-window {
  background-color: #377d73;
  border-radius: 0.3em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 300px;
  height: 500px;
}

.emotion-1 .login-window .text {
  margin-bottom: 30px;
}

.emotion-0 {
  cursor: pointer;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 130px;
  height: 30px;
  background-color: #2b2d2f;
  border-radius: 7px;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);
  -webkit-transition: all 0.3s cubic-bezier(0.25,0.8,0.25,1);
  transition: all 0.3s cubic-bezier(0.25,0.8,0.25,1);
  overflow: hidden;
}

.emotion-0 *,
.emotion-0 *:before,
.emotion-0 *:after {
  box-sizing: border-box;
  outline: none;
}

.emotion-0:before,
.emotion-0:after {
  position: absolute;
  top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 50%;
  height: 100%;
  -webkit-transition: 0.25s linear;
  transition: 0.25s linear;
  z-index: 1;
}

.emotion-0:before {
  content: '';
  left: 0;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  background-color: #2b2d2f;
}

.emotion-0:after {
  content: '';
  right: 0;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  background-color: #2b2d2f;
}

.emotion-0:hover {
  background-color: #fff;
  box-shadow: 0 14px 28px rgba(0,0,0,0.25),0 10px 10px rgba(0,0,0,0.22);
}

.emotion-0:hover span {
  opacity: 0;
  z-index: -3;
}

.emotion-0:hover:before {
  opacity: 0.5;
  -webkit-transform: translateY(-100%);
  -ms-transform: translateY(-100%);
  transform: translateY(-100%);
}

.emotion-0:hover:after {
  opacity: 0.5;
  -webkit-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
}

.emotion-0 span {
  position: absolute;
  top: 0;
  left: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  color: #f5f5f5;
  font-size: 12px;
  font-weight: 700;
  opacity: 1;
  -webkit-transition: opacity 0.25s;
  transition: opacity 0.25s;
  z-index: 2;
}

.emotion-0 .social-link {
  cursor: pointer;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  color: #f5f5f5;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: 0.25s;
  transition: 0.25s;
}

.emotion-0 .social-link svg {
  width: 20%;
  height: 100%;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.emotion-0 .social-link:hover {
  background-color: rgba( #f5f5f5, 0.1 );
  -webkit-animation: animation-0 0.4s linear;
  animation: animation-0 0.4s linear;
}

<div
  className="emotion-1"
>
  <div
    className="login-window"
  >
    <div
      className="text"
    >
      Please login to continue
    </div>
    <div
      className="emotion-0"
    >
      <span>
        Login With Google
      </span>
      <div
        className="social-link"
        onClick={[Function]}
      >
        <svg
          viewBox="0 0 533.5 544.3"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M533.5 278.4c0-18.5-1.5-37.1-4.7-55.3H272.1v104.8h147c-6.1 33.8-25.7 63.7-54.4 82.7v68h87.7c51.5-47.4 81.1-117.4 81.1-200.2z"
            fill="#4285f4"
          />
          <path
            d="M272.1 544.3c73.4 0 135.3-24.1 180.4-65.7l-87.7-68c-24.4 16.6-55.9 26-92.6 26-71 0-131.2-47.9-152.8-112.3H28.9v70.1c46.2 91.9 140.3 149.9 243.2 149.9z"
            fill="#34a853"
          />
          <path
            d="M119.3 324.3c-11.4-33.8-11.4-70.4 0-104.2V150H28.9c-38.6 76.9-38.6 167.5 0 244.4l90.4-70.1z"
            fill="#fbbc04"
          />
          <path
            d="M272.1 107.7c38.8-.6 76.3 14 104.4 40.8l77.7-77.7C405 24.6 339.7-.8 272.1 0 169.2 0 75.1 58 28.9 150l90.4 70.1c21.5-64.5 81.8-112.4 152.8-112.4z"
            fill="#ea4335"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
